core/domain/recommendations_jobs_one_off_test.py

```diff
@@ -97,3 +98,29 @@ def test_recommendations_after_changes_in_rights(self):
                 recommendations_services.get_exploration_recommendations(
                     'exp_id_1'))
             self.assertEqual(recommendations, ['exp_id_2', 'exp_id_3'])
+
+    def test_recommendations_with_invalid_exploration_summaries(self):
+
+        def _mock_get_non_private_exploration_summaries():
```

Comment:
From just reading this file, I don't really understand this. What do non-private exploration summaries have to do with recommendations? The two concepts don't seem obviously linked and I am not sure I understand the rationale for this test, so at the very least there should be an explanation for it in comments.

Reply:
I am actually testing [this line](https://github.com/oppia/oppia/blob/develop/core/domain/recommendations_jobs_one_off.py#L58). I created a mock function that returns a different exp_summaries_dict so that the recommendations would be an empty list (since`get_exploration_recommendations` returns a list of ids of at most 10 recommended explorations to play after completing the exploration)

Reply:
I see. You should explain this in a line comment, so that future devs aren't confused. 

Also, please update the name of the test to include "what should happen" -- what behaviour are we expecting to occur. See point 2 [here](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests). That should help make things clearer.

Reply:
Done